frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.unit.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import { screen } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+import _ from "underscore";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders } from "__support__/ui";
+import { Card, Collection } from "metabase-types/api";
+import {
+  createMockCard,
+  createMockNativeDatasetQuery,
+} from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import { createMockState } from "metabase-types/store/mocks";
+import { checkNotNull } from "metabase/core/utils/types";
+import { getMetadata } from "metabase/selectors/metadata";
+import NativeQuery from "metabase-lib/queries/NativeQuery";
+
+const { NativeQueryEditor } = jest.requireActual(
+  "metabase/query_builder/components/NativeQueryEditor",
+);
+
+const TEST_DB = createSampleDatabase();
+
+const TEST_NATIVE_CARD = createMockCard({
+  dataset_query: createMockNativeDatasetQuery({
+    database: TEST_DB.id,
+  }),
+});
+
+const ROOT_COLLECTION = {
+  id: "root",
+  name: "Our analytics",
+  can_write: true,
+} as Collection;
+
+interface SetupOpts {
+  card?: Card;
+  height?: number;
+  isActive: boolean;
+  readOnly?: boolean;
+}
+
+const setup = async ({
+  card = TEST_NATIVE_CARD,
+  height = 300,
+  isActive,
+  readOnly = false,
+}: SetupOpts) => {
+  setupDatabasesEndpoints([TEST_DB]);
+
+  const storeInitialState = createMockState({
+    entities: createMockEntitiesState({
+      databases: [createSampleDatabase()],
+      questions: [card],
+    }),
+  });
+  const metadata = getMetadata(storeInitialState);
+  const query = new NativeQuery(
```

Comment:
I might be missing something but it seems that this can be simplified. You should be able to set `dataset_query` in `TEST_CARD`, and then `const question = question.query()` without constructing the query manually (which can be incorrect)

Reply:
ad1ed026b09d0a0761cfafcfff65ffe0496aea67